components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/OAuth2ScopeService.java

```diff
@@ -823,4 +829,32 @@ private UserApplicationScopeConsentDO getConsentsToBeAdded(UserApplicationScopeC
         consentToBeAdded.setDeniedScopes(disapprovedScopes);
         return consentToBeAdded;
     }
+
+    @Override
+    public String getName() {
+
+        return "OAuth2ScopeService";
+    }
+
+    @Override
+    public List<OAuth2Resource> getMetadata(List<String> scopes) {
+
+        List<ScopeMetadata> scopesArray = new ArrayList<>();
+        for (String scopeName : scopes) {
+            try {
+                Scope scope = getScope(scopeName);
+                ScopeMetadata scopeMetadata = new ScopeMetadata(scope.getName(), scope.getDisplayName(),
+                        scope.getDescription());
+                scopesArray.add(scopeMetadata);
+            } catch (IdentityOAuth2ScopeException e) {
+                log.error("Error while retrieving scope: " + scopeName);
```

Comment:
Log the exception (e). Whats the reason for not breaking the flow and propagating the exception?

Reply:
Since the scope metadata is not critical for the authentication flow, we can gracefully handle the error thrown here.